### ISSUE TITLE
Not compatible with Elastic Suite 2.9.0?

### ISSUE THREAD

**Issue by trebb0r** [Smile-SA/magento2-module-elasticsuite-shared-catalog#4]
Or not needed anymore since 2.9.0?

**Comment by davidstillson** [comment]
I second this request/bug..  can't install 2.9.0 because of composer dependency conflicts

**Comment by a-dite** [comment]
Same here
`Problem 1
    - Installation request for smile/module-elasticsuite-shared-catalog ^1.1.0 -> satisfiable by smile/module-elasticsuite-shared-catalog[1.1.0].
    - Can only install one of: smile/elasticsuite[2.9.0, 2.8.x-dev].
    - Can only install one of: smile/elasticsuite[2.9.x-dev, 2.8.x-dev].
    - Conclusion: install smile/elasticsuite 2.8.x-dev
    - Installation request for smile/elasticsuite ~2.9.0 -> satisfiable by smile/elasticsuite[2.9.0, 2.9.x-dev].
`

**Comment by duffner** [comment]
@romainruaud, can you please take a look at this? This is an issue for us as well

**Comment by trebb0r** [comment]
Or maybe @androshchuk or anyone at Smile that can help out?

**Comment by rbayet** [comment]
Hello guys,

Until we find a proper solution, have you tried cheating with composer and installing Elasticsuite with `composer require smile/elasticsuite 2.9.0 as 2.8.7` for instance ?
I know I used this trick when I wanted a M2 2.3.5 / Elasticsuite 2.9.0 with https://github.com/Smile-SA/elasticsuite-for-retailer.

(I offer no guarantee of compatibility yet, but this solves the composer issue).

Regards,

**Comment by actionis** [comment]
Will this be compatible with 2.9.0?

Is it just a matter of updating the requirements for `smile/elasticsuite` in `composer.json`?

**Comment by skukla** [comment]
Is there any chance of these modules being updated?  Given that ES 6.x is now deprecated in Magento 2.4.0 (although not yet officially), it would be great to be able to use the most up-to-date version of Elasticsuite together with Magento's native B2B functionality.

**Comment by romainruaud** [comment]
Guys,

I don't want to be rude but please keep in mind that we are providing all of these modules **for free**, and on an **Open Source License**. This means that **you** can also participate and submit pull requests to make this module evolve as fast as you need.

If you only want to take benefits of it without helping, that's fine too, but you'll have to accept that we may have some delay.

Regards

**Comment by davidstillson** [comment]
@romainruaud Not rude at all, it's good to know that you will accept PRs for this..  What is the typical turn around for you guys to code review/release?   Is there a preferred process or format for any PRs submitted?

**Comment by skukla** [comment]
As @davidstillson points out, not rude at all, and I apologize if my comment came off harshly. Smile is one of the best agencies I have seen at responding to requests like this and I took that for granted in this case. Seeing others ask whether this was supported in 2.9.x had me concerned that support for B2B might have been dropped. 

@romainruaud, Thanks for the update and as always, thanks to you and the Smile team for the effort and work on this product.

**Comment by romainruaud** [comment]
You can just create a pull request pointing to **master**, this repository is tiny and does not have all the complex validation rules (Travis builds, unit testing etc...) of the ElasticSuite core repository.

I'll gladly review it after my vacations, but in any cases, I think the composer trick given by @rbayet could help you installing the module at least. I don't know however if this would be fully functional.

Regards

**Comment by skukla** [comment]
Hi all.  Given that I'm not a developer, it's often difficult for me to find ways to contribute back to projects like this; however, I've just done some testing and thought it might be helpful if I reported back what I learned:

1. @rbayet's suggestion of using a composer alias _does_ allow for the B2B modules to be installed with the latest version of Elasticsearch (~2.10.0); however, in my testing, this only works when one includes the require statement in one's `composer.json` as opposed to an inline composer require statement.  (E.g. running the following from the command line produced an error: `composer require smile/elasticsuite 2.10.0 as 2.8.7`; but using it in my composer.json file worked fine).
2. With the modules installed and the proper search engine configured, Elasticsuite works: results are returned via typeahead.  However, product image thumbnails are _not_ included in the typeahead results.

That's as far as I've dug in thus far.  Hopefully it's helpful.

**Comment by adamchukandrii** [comment]
Hello, @romainruaud. Is there any news on this issue?

**Comment by romainruaud** [comment]
I just came back from vacations, I'll try to find some time this week to release a new version of this module.

Regards

**Comment by adamchukandrii** [comment]
Thanks

**Comment by romainruaud** [comment]
Please try the 1.1.1 version, which is intended to allow Elasticsuite 2.8 or 2.9 with Magento 2.3.x versions.

If that's okay I'll also release a version allowing Elasticsuite 2.10 for Magento 2.4.x versions.

Thanks

**Comment by adamchukandrii** [comment]
Looks like all is ok. Thank you

**Comment by rbayet** [comment]
@romainruaud Should we close this one ?

**Comment by rbayet** [comment]
Hello @skukla (and others),

I've just published a 2.10.0 release which is compatible afaik with Magento 2.4.1 / Elasticsuite 2.10.2 / B2B features 1.3.x.
It simply contain a different Elasticsuite requirement.

Same thing for new releases 2.6.0, 2.8.0 and 2.9.0 releases which are strictly the same thing code-wise as, respectively, releases 1.0.1, 1.1.0 and 1.1.1.

We've decided to adopt the same pattern of version numbers as Elasticsuite for this module and for the [Elasticsuite Quick Order module](https://github.com/Smile-SA/magento2-module-elasticsuite-quick-order/) to help identify Magento2 / Elasticsuite compatibility.
I have no illusion that Magento is totally capable of punching us in the face real hard with the next minor 2.3.x and 2.4.x releases, but where's the fun otherwise ? :)

Regards,
